datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/exception/DataHubDataFetcherExceptionHandler.java

```diff
@@ -34,6 +34,14 @@ public CompletableFuture<DataFetcherExceptionHandlerResult> handleException(
       message = illException.getMessage();
     }
 
+    IllegalStateException illStateException =
+        findFirstThrowableCauseOfClass(exception, IllegalStateException.class);
+    if (illStateException != null) {
+      log.error("Failed to execute", illStateException);
+      errorCode = DataHubGraphQLErrorCode.SERVER_ERROR;
```

Comment:
I could see illegal state mapping to a 4xx maybe as well, depends on how we're using it